proto/planning/api.proto

```diff
@@ -1,131 +1,17 @@
-message AngleBetweenVectorsConstraint {
-  string frame_A = 1;
-  string frame_B = 2;
-  repeated double a_A = 3;
-  repeated double b_B = 4;
-  double angle_lower = 5;
-  double angle_upper = 6;
-}
-// Collection of all constraints enforced on the given plan
-message Constraints {
-  repeated PositionConstraint pos_constraints = 1;
-  repeated AngleBetweenVectorsConstraint angle_constraints = 2;
-  // TODO(@davebambrick): Add collision checking
-}
-
-// Joint limits for each joint in the system
-message JointLimits {
-  repeated double lower = 1;
-  repeated double upper = 2;
-}
-
-// placeholder message type for "new" params (to be
-// used by GCS)
-message Params {
-  string system_name = 1;
-  string model_directive = 2;
-  string robot_name = 3;
-  string link_name = 4;
-  repeated double link_axis = 5;
-  repeated double world_axis = 6;
-  JointLimits joint_position_limits = 7; // need convert to std::pair<VectorX<double>, VectorX<double>>
-
-  // planning params
-  double max_solve_time = 8;
-  double position_tolerance = 9;
-  double orientation_tolerance = 10;
-  double collision_buffer_meters = 11;
-  bool check_plan_validity = 12;
-  double collision_cost_scale = 13;
-  double orientation_cost_scale = 14;
-  double stretch_factor = 15;
-  double dense_delta = 16;
-  bool dense_roadmap = 17;
-  double heuristic_scaling = 18;
 
-  // recall from database
-  bool recall = 19;
-  double recall_time_cap_seconds = 20;
-  bool collision_check_on_recall = 21;
-}
-
-// system polynomial; a map of individual
-// polynomials to their respective target robots
-message SysPoly {
-  map<string, Poly> data = 1;
-}
-
-// individual piecewise polynomial trajectory, uniquely defined by
-//   1) an array of "breaks", or points in time representing
-//      the intervals for constituent polynomials.
-//   2) an array of m-by-n matrices of coeffecient vectors, where
-//      a single constituent matrix uniquely defines a given
-//      polynomial. each matrix contains the coefficients of the
-//      underlying additive monomials.
-// note: the coeffecients are stored in this message type
-// as a vector of vectors, so care must be taken to properly
-// unpack this data into the appropriate structure
-message Poly {
-  repeated double breaks = 1;
-  uint32 rows = 2;
-  uint32 cols = 3;
-  // vector of monomial coefficients
-  repeated Coeffs coeffs = 4;
-}
-
-// a vector of coefficients for a single monomial
-message Coeffs {
-  repeated double data = 1;
-}
-
-// request sent by a client to start a given plan
+/** Request containing all information necessary to uniquely define a planning
+ * problem in a format accepted by the planner. */
 message StartPlanRequest {
   string id = 1;
   ProblemDef def = 2;
-  Params params = 3;
-  string legacy_params = 4; // TODO(@davebambrick): deprecate
-  Constraints constraints = 5;
+  string legacy_params = 3;  // TODO(@davebambrick): deprecate
```

Comment:
Does changing the index of the field break backwards compatibility? Do we care about that right now?

Reply:
My opinion is, not until we officially deploy. I _do_ think after this change we should consider effectively disallowing deletions.